pontoon/sync/formats/json_extensions.py

```diff
@@ -0,0 +1,187 @@
+"""
+Parser for the .json translation format as used by the WebExtensions API:
+https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Internationalization
+
+See also:
+https://www.chromium.org/developers/design-documents/extensions/how-the-extension-system-works/i18n
+"""
+import codecs
+import copy
+import json
+import logging
+import os
+
+from collections import OrderedDict
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+from pontoon.sync import SyncError
+from pontoon.sync.exceptions import ParseError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+
+log = logging.getLogger(__name__)
+
+SCHEMA = {
+    "type": "object",
+    "patternProperties": {
```

Comment:
im wondering if this would be simpler as just properties without the pattern match

Reply:
I'm not sure how to do that. The JSON file contains a [list of objects](https://github.com/mathjazz/pontoon-test/blob/master/en-US/example.json) where anything can be a key.

Reply:
just use `properties`

Reply:
aplogies - above comments should have said `additionalProperties` rather than `properties`